src/config.rs

```diff
@@ -4,16 +4,36 @@ use std::result::Result;
 use std::io::{Error};
 use std::path::{Path};
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
```

Comment:
Why have we picked up `Clone` here? I'm sure there's a good reason it's just not jumping out at me from the code